### DISCUSSION TITLE
Arduino Uno readDigital equivalent

### DISCUSSION THREAD

**Discussion by jbugalho** [Rahix/avr-hal#511] [Q&A] [answered]
Hello, I'm trying to learn to use Arduino with Rust, and I have been trying to do a small traffic light when a button is inserted. For some reason this code is always lighting the green led! Can someone help me?

```RUST
#![no_std]
#![no_main]

use panic_halt as _;

#[arduino_hal::entry]
fn main() -> ! {
    let dp = arduino_hal::Peripherals::take().unwrap();
    let pins = arduino_hal::pins!(dp);

    let mut red = pins.d12.into_output();
    let mut yellow = pins.d11.into_output();
    let mut green = pins.d10.into_output();
    let button = pins.d9; // i think the default is input (?)

    loop {
        green.set_high();
        if button.is_high() {
            green.set_low();
            yellow.set_high();
            arduino_hal::delay_ms(2000);
            yellow.set_low();
            red.set_high();
            arduino_hal::delay_ms(1000);
            red.set_low();
        }
    }
}
```

**Comment by Rahix** [comment, marked as answer]
(Converted this to a discussion)

> ```rust
>  let button = pins.d9; // i think the default is input (?)
> ```

Yes, default is input.  Actually, you can only call the methods for input pins when the pin is configured as input, so you can't accidentally do the wrong thing here :)

> For some reason this code is always lighting the green led!

Well, that's exactly what you wrote in your code?  While the button is not pressed, it will continuously loop over setting the green LED pin high and then checking whether the button pin is high.

Maybe what you mean is that pressing the button doesn't do anything? If yes, that has to be an electrical issue.  How did you connect the button?

**Reply by jbugalho** [reply]
Ty, for some reason i spent an half hour looking for the issue in code and couldnt find that I had defined port 9 for input button but the cabble was on port 2 xD ty anyways
